### PR TITLE
improve `localnet.md`

### PR DESCRIPTION
**Pull request by raulk**
- Details system prerequisites (e.g. Docker, Rust, etc.) and provides instructions for Linux (Ubuntu).
- Adds missing details for when users start from scratch (which is the most probable scenario).
- Add some explanations and clean up.

### CODE REVIEW

File: docs/localnet.md [hunk 1]
```diff
@@ -21,41 +39,48 @@ cargo make --makefile ./infra/Makefile.toml testnode-down
...
-- Run CometBFT in 4 containers
-- Run Eth API in 4 containers
\ No newline at end of file
+## What's happening behind the scenes
+
+> For a 4-node deployment.
+
+The deployment process performs the following steps:
+
+- Remove all Docker containers, files, networks, etc. from any previous deployments.
```
Comment (adlrocha) [inline docs/localnet.md:78]: The scripts should be handling this for you, but definitely useful as a sanity-check

File: docs/localnet.md [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
+This guide offers two flavours:
+
+- A single node deployment: useful for developing smart contracts and testing the APIs.
+- A 4 node testnet: useful for testing consensus, checkpointing, and more.
+
+## Prerequisites
+
+On Linux (links and instructions for Ubuntu):
+
+- Install system packages: `sudo apt install build-essential clang cmake pkg-config libssl-dev protobuf-compiler`.
```
Comment (adlrocha) [inline docs/localnet.md:14]: We should maybe add this to a `make install-deps` to make users lives easier. I will track it in an issue.

**Review by adlrocha — APPROVED** [review]
This was long overdue. Thank you @raulk.

**Comment by raulk** [comment]
Merged this branch into #329. Closing in favour of that.

File: docs/localnet.md (hunk 2 above)
Comment (raulk) [reply docs/localnet.md:14]: Can do, but this step is OS and distribution-dependent.

- Would need to check compatibility and abort if an unknown/unsupported Linux distro is being used.
- Would be great to make it support macOS.

Having said that, I'm migrating these scripts to use docker images, so this step will soon become obsolete/unnecessary.

**Pull request closed without merging**